vector/src/main/java/im/vector/app/features/home/room/list/RoomListFragment.kt

```diff
@@ -142,6 +151,33 @@ class RoomListFragment @Inject constructor(
                         (it.contentEpoxyController as? RoomSummaryPagedController)?.roomChangeMembershipStates = ms
                     }
         }
+
+        homeSharedActionViewModel
+                .stream()
+                .mapNotNull { it as? HomeActivitySharedAction.SelectTab }
+                .onEach { handleSelectTab(it.tab) }
+                .launchIn(viewLifecycleOwner.lifecycleScope)
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
+        super.onCreateOptionsMenu(menu, inflater)
+
+        val searchItem = menu.findItem(R.id.menu_home_search_action)
+        val searchView = searchItem?.actionView as? SearchView ?: return
+
+        // For initial filter
+        withState(roomListViewModel) { state ->
+            if (state.roomFilter.isNotEmpty()) {
+                searchItem.expandActionView()
+                searchView.setQuery(state.roomFilter, true)
+                searchView.clearFocus()
+            }
+        }
+
+        searchView.queryTextChanges()
+                .debounce(300)
```

Comment:
why 300 ? Magic number :)
Maybe you can use a ```private const val ```

Reply:
I agree about the magic number, I got it from other places in the code... The goal is to reduce requests and UI updates, I am not sure it will be more relevant if we move it to a const